### PR TITLE
Build both core agent and single binary for Heroku

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

This change modifies the Heroku package so that it includes:
- a binary for the core agent
- a binary for core agent + process agent

### Motivation

Some customers do not use the process agent, in this case the process-agent
binary is removed from the Heroku buildpack. When using a single binary, they
end up having a bigger binary.

### CODE REVIEW

File: tasks/agent.py [hunk 1]
```diff
@@ -128,6 +128,7 @@ def build(
     cmake_options='',
     bundle=None,
     bundle_ebpf=False,
+    agent_bin='',
```
Comment (chouquette) [inline tasks/agent.py:131]: I'd feel more at ease with `None` as the default value, it should fail in a clearer way in case we end up using it as is.

**Review by chouquette — APPROVED** [review]
Looks OK!

I'm curious as to why you need a new `agent_bin` parameter specifically for heroku though!

File: tasks/agent.py (hunk 1 above)
Comment (dustmop) [reply tasks/agent.py:131]: Agree on this point. Both are falsey values, but using `None` makes the intent stand out more.

**Comment by lebauce** [comment]
@chouquette The reason is that I want to create multiple agent binaries and this sounds like a cleaner way than renaming files

**Review by dustmop — APPROVED** [review]
LGTM

**Pull request merged**